### DISCUSSION TITLE
Path params:``value is not a valid integer`` why i get this?

### DISCUSSION THREAD

**Discussion by dongruixiao** [fastapi/fastapi#11572] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()

@app.get("/a/{a}/b/{b}/c/{c}")
def read_root(a:str, b: str, c: int):
    return {"a":a, "b": b, "c": c}

a = app.openapi()
print(a)
```

### Description

run this app with ``uvicorn app:app``

i use this path params:
```
a: "a"
b: "b"
c: 123
```

i get this:
```
INFO:     Started server process [75144]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
INFO:     127.0.0.1:58050 - "GET /docs HTTP/1.1" 200 OK
INFO:     127.0.0.1:58050 - "GET /openapi.json HTTP/1.1" 200 OK
INFO:     127.0.0.1:58111 - "GET /a/a/b/%7Bb%7D/c/%7Bc%7D HTTP/1.1" 422 Unprocessable Entity
```
i think the path params is not parser right?

but when i open this url in my brower : 

```http://127.0.0.1:8000/a/a/b/b/c/123``` 

its return very well
```
{
a: "a",
b: "b",
c: 123
}
```
in the swagger ui i found that:

```
curl -X 'GET' \
  'http://127.0.0.1:8000/a/a/b/{b}/c/{c}' \
  -H 'accept: application/json'
```
it not parse my `b` with `b` and `c` with `123`
it pass `b` with `{b}` and `c` with `{c}`

and then got this error:

```
{
  "detail": [
    {
      "loc": [
        "path",
        "c"
      ],
      "msg": "value is not a valid integer",
      "type": "type_error.integer"
    }
  ]
}
```

and this is the openapi schema:
```
{'openapi': '3.1.0',
 'info': {'title': 'FastAPI', 'version': '0.1.0'},
 'paths': {'/a/{a}/b/{b}/c/{c}': {'get': {'summary': 'Read Root',
    'operationId': 'read_root_a__a__b__b__c__c__get',
    'parameters': [{'required': True,
      'schema': {'type': 'string', 'title': 'A'},
      'name': 'a',
      'in': 'path'},
     {'required': True,
      'schema': {'type': 'string', 'title': 'B'},
      'name': 'b',
      'in': 'path'},
     {'required': True,
      'schema': {'type': 'integer', 'title': 'C'},
      'name': 'c',
      'in': 'path'}],
    'responses': {'200': {'description': 'Successful Response',
      'content': {'application/json': {'schema': {}}}},
     '422': {'description': 'Validation Error',
      'content': {'application/json': {'schema': {'$ref': '#/components/schemas/HTTPValidationError'}}}}}}}},
 'components': {'schemas': {'HTTPValidationError': {'properties': {'detail': {'items': {'$ref': '#/components/schemas/ValidationError'},
      'type': 'array',
      'title': 'Detail'}},
    'type': 'object',
    'title': 'HTTPValidationError'},
   'ValidationError': {'properties': {'loc': {'items': {'anyOf': [{'type': 'string'},
        {'type': 'integer'}]},
      'type': 'array',
      'title': 'Location'},
     'msg': {'type': 'string', 'title': 'Message'},
     'type': {'type': 'string', 'title': 'Error Type'}},
    'type': 'object',
    'required': ['loc', 'msg', 'type'],
    'title': 'ValidationError'}}}}
```
i also change the openapi version to 3.0.2 
it not work also

why i get this?
can u help me?
😭

### Operating System

macOS

### FastAPI Version

0.99.1

### Pydantic Version

1.10.15

### Python Version

Python 3.11.3

**Comment by YuriiMotov** [comment]
I tried your code example and it works as expected.
Can you clarify what you mean saying

> i use this path params:

> a: "a"
> b: "b"
> c: 123

How do you pass these params?


`http://127.0.0.1:8000/a/a/b/b/c/123` is a right way to pass these params when you do a API call using web-browser. And if you use Swagger UI, you should just type parameters into fields without any curly braces of quote signs.

**Reply by dongruixiao** [reply]
thanks for your reply

in short, i open this url ```http://127.0.0.1:8000/a/a/b/b/c/123``` is well，
but i use swagger ui it got this:

<img width="442" alt="image" src="https://github.com/tiangolo/fastapi/assets/16684015/46730803-c188-4a50-bcf9-f78b6cd8edfd">

**Comment by YuriiMotov** [comment, marked as answer]
There is a bug in current version of Swagger UI (https://github.com/swagger-api/swagger-ui/issues/9928).
Latest version of FastAPI pins Swagger UI version to 5.9.0. But FastAPI 0.99.1 (which you use) uses latest version of Swagger UI, which contains mentioned bug.

To solve your problem you can pin Swagger UI version to 5.9.0 adding the following code to your project:

```
app = FastAPI(docs_url=None)  # Disable default docs

from fastapi.openapi.docs import (get_swagger_ui_html,
                                  get_swagger_ui_oauth2_redirect_html)

@app.get("/docs", include_in_schema=False)
async def custom_swagger_ui_html():
    return get_swagger_ui_html(
        openapi_url=app.openapi_url,
        title=app.title + " - Swagger UI",
        oauth2_redirect_url=app.swagger_ui_oauth2_redirect_url,
        swagger_js_url="https://unpkg.com/swagger-ui-dist@5.9.0/swagger-ui-bundle.js",
        swagger_css_url="https://unpkg.com/swagger-ui-dist@5.9.0/swagger-ui.css",
    )
```

Full code example:
```
from fastapi import FastAPI

app = FastAPI(docs_url=None)


from fastapi.openapi.docs import (get_swagger_ui_html,
                                  get_swagger_ui_oauth2_redirect_html)


@app.get("/docs", include_in_schema=False)
async def custom_swagger_ui_html():
    return get_swagger_ui_html(
        openapi_url=app.openapi_url,
        title=app.title + " - Swagger UI",
        oauth2_redirect_url=app.swagger_ui_oauth2_redirect_url,
        swagger_js_url="https://unpkg.com/swagger-ui-dist@5.9.0/swagger-ui-bundle.js",
        swagger_css_url="https://unpkg.com/swagger-ui-dist@5.9.0/swagger-ui.css",
    )

@app.get(app.swagger_ui_oauth2_redirect_url, include_in_schema=False)
async def swagger_ui_redirect():
    return get_swagger_ui_oauth2_redirect_html()


@app.get("/a/{a}/b/{b}/c/{c}")
def read_root(a:str, b: str, c: int):
    return {"a":a, "b": b, "c": c}

a = app.openapi()
print(a)

```

**Reply by dongruixiao** [reply]
tks sir, you did a great job
